components/ws-daemon/pkg/cgroup/plugin_iolimit_v1.go

```diff
@@ -55,80 +64,49 @@ const (
 // TODO: enable custom configuration
 var blockDevices = []string{"sd*", "md*", "nvme0n*"}
 
-func (c *IOLimiterV1) Update(writeBytesPerSecond, readBytesPerSecond, writeIOPs, readIOPs int64) {
-	c.cond.L.Lock()
-	defer c.cond.L.Unlock()
-
-	c.limits = limits{
+func buildLimits(writeBytesPerSecond, readBytesPerSecond, writeIOPs, readIOPs int64) limits {
+	return limits{
 		WriteBytesPerSecond: writeBytesPerSecond,
 		ReadBytesPerSecond:  readBytesPerSecond,
 		WriteIOPS:           writeIOPs,
 		ReadIOPS:            readIOPs,
+
+		cache:   make(map[string][]string),
```

Comment:
As far as I know, `IOLimiterV1` is a singleton. It means this `cache` is shared by all instances managed by a given ws-damon. Is this as intended?

Reply:
Yes. We don't have different configurations per workspace

Reply:
It would be good to comment on why that is okay. Perhaps, but because ws-daemon is on each node, right?